src/components/application_manager/src/commands/mobile/perform_audio_pass_thru_request.cc

```diff
@@ -179,36 +177,51 @@ void PerformAudioPassThruRequest::on_event(const event_engine::Event& event) {
     return;
   }
 
-  std::string return_info;
-  mobile_apis::Result::eType result_code = mobile_apis::Result::INVALID_ENUM;
-  const bool result = PrepareResponseParameters(result_code, return_info);
+  ResponseParams response_params = PrepareResponseParameters();
 
-  SendResponse(result,
-               result_code,
-               return_info.empty() ? NULL : return_info.c_str(),
-               &(message[strings::msg_params]));
+  SendResponse(
+      response_params.result,
+      response_params.result_code,
+      response_params.info.empty() ? NULL : response_params.info.c_str(),
+      &(message[strings::msg_params]));
 }
 
-bool PerformAudioPassThruRequest::PrepareResponseParameters(
-    mobile_apis::Result::eType& result_code, std::string& info) {
+const PerformAudioPassThruRequest::ResponseParams&
+PerformAudioPassThruRequest::PrepareResponseParameters() {
```

Comment:
@LuxoftAKutsan 
I'm not sure whether implementation done in base class is good enough however I think we have to have some common logic for all command as there are common requirements. So in case you'll be overriding and extending logic of base class later it may be easier to come up with improved solution and do some refactoring since it will be easier to understand the code and find similar places. And as opposite - if you'll be creating bunch of local methods in different commands which in fact do almost same checks it will be lot harder to separate common logic so we'll end up with number of duplicated code and logic.